ICSharpCode.Decompiler/TypeSystem/Implementation/MetadataField.cs

```diff
@@ -42,7 +42,7 @@ sealed class MetadataField : IField
 		object constantValue;
 		IType type;
 		bool isVolatile; // initialized together with this.type
-		byte decimalConstant; // 0=no, 1=yes, 2=unknown
+		bool? isDecimalConstant;
```

Comment:
This change is not thread-safe.
In the rare case where multiple threads are concurrently initializing `isDecimalConstant`, we need accesses to be atomic.
But `Nullable<T>` is a struct and may have torn reads/writes.